### ISSUE TITLE
Question regarding client TS generation and nullable query params

### ISSUE THREAD

**Issue by arnederuwe** [RSuter/NSwag#2000]
Hi,

Consider the following API method:
```public async Task<ActionResult<AssignmentDto>> GetAssignmentById(Guid? id)```

NSwag generates the following TS code:
```
getAssignmentById(id: string | null | undefined): Promise<AssignmentDto | null> {
        let url_ = this.baseUrl + "/api/Assignment/id?";
        if (id !== undefined)
            url_ += "id=" + encodeURIComponent("" + id) + "&"; 
        url_ = url_.replace(/[?&]$/, "");

       ...
    }
```

Why doesn't it generate:
```
getAssignmentById(id: string | null | undefined): Promise<AssignmentDto | null> {
        let url_ = this.baseUrl + "/api/Assignment/id?";
        if (id != null) // <==
            url_ += "id=" + encodeURIComponent("" + id) + "&"; 
        url_ = url_.replace(/[?&]$/, "");

       ...
    }
```

Is there a reason to let 'null' pass here? Web API sure doesn't like it, the modelbinder fails when passing a query parameter with null as its value.

**Comment by RSuter** [comment]
The "id" parameter is nullable and optional - so the idea here is to not send it when undefined and send null to actually specify null - but i think the parameter model binder cannot handle ?id=null right?

**Comment by arnederuwe** [comment]
Exactly, .NET Core model binding errors on ?id=null
I get there's a difference with passing nothing and passing ?id=null, but in the case of .NET, I can't think of any scenario where you would want to pass ?id=null instead of nothing at all. And if you did want to, you would have to write your own model binder for it as it isn't supported OOTB.

**Comment by RSuter** [comment]
It’s important to say that this is NOT about asp.net core... ie we should not handle null so that it works with asp.net core but how it is meant with the openapi spec... but what would you generate? id= or nothing?

**Comment by arnederuwe** [comment]
I would be in favor for generating nothing, but I'm not sure if there is some spec out there that dictates otherwise (I couldn't find it by googling).
I would choose generating nothing because 'param=' might translate to an empty string instead of null in some frameworks, which seems valid

**Comment by RSuter** [comment]
I’ll check with the asp team.. one question is:
Assuming you have this query param: int? foo = 5
How can you pass null? Providing nothing would be the default (5) and enpty string might be null... but as you said this wont work with string (empty vs null).

**Comment by arnederuwe** [comment]
Good point.. I guess the ideal way to resolve this would be that .NET Core **does** parse 'null' in the query string...
Did you check with MS on github? I would like to see that discussion

**Comment by RSuter** [comment]
@pranavkm can you give some insights on how this is handled in ASP.NET Core and what should be used in as query parameter data?

**Comment by srollinet** [comment]
Any news about this issue? I am facing the same problem with nullable int

> Good point.. I guess the ideal way to resolve this would be that .NET Core does parse 'null' in the query string...
But then, how do you represent a string with the value "null" ?

**Comment by RicoSuter** [comment]
I think null cannot be provided as query param
